src/Utilities/StdArrayHelpers.hpp

```diff
@@ -26,6 +31,9 @@ inline std::array<T, Dim>& operator+=(std::array<T, Dim>& lhs,
   return lhs;
 }
 
+// triggered e.g. when used with int and size_t
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsign-conversion"
```

Comment:
Ignoring this warning seems a bit dangerous :) Is it to allow `T` and `U` have different signedness? Perhaps just leave the warning here and silence it in calling code if necessary? Or use `static_cast`s here?

Reply:
yes, I dont know what changed in clang-tidy that this triggers now. It is triggered by `Test_StdArrayHelpers.cpp`, where the import 
```
// We wish to explicitly test implicit type conversion when adding std::arrays
// of different fundamentals, so we supress -Wsign-conversion.
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wsign-conversion"
#include "Utilities/StdArrayHelpers.hpp"
#pragma GCC diagnostic pop
```
is supposed to prevent it. This doesnt work anymore and I dont know why, any ideas?

Reply:
ah, I figured it out and moved this to the top of the include statements. presumably some other header now includes it as well and the `pragma once` prevents the include within the prepro silencer